docs-ui/components/sample.tsx

```diff
@@ -1,12 +1,79 @@
+import {createContext, ReactChild, useState} from 'react';
+import {ThemeProvider, useTheme} from '@emotion/react';
 import styled from '@emotion/styled';
 
+import {IconMoon} from 'app/icons';
 import space from 'app/styles/space';
+import {darkTheme, lightTheme, Theme} from 'app/utils/theme';
 
-const Sample = styled('div')`
+type ThemeName = 'dark' | 'light';
+
+type Props = {
+  children?: ReactChild;
+  showThemeSwitcher?: boolean;
+  noBorder?: boolean;
+};
+
+/** Expose the selected theme to children of <Sample /> */
+export const SampleThemeContext = createContext<ThemeName>('light');
```

Comment:
Exporting `SampleThemeContext` so children of `Sample />` can consume it (via `useContext(SampleThemeContext)`) for their own purposes.

Reply:
Hmm why would children need to use this value?

Reply:
In the Colors page that I'm working on, there are color samples for both light and dark mode:
<img width="647" alt="Screen Shot 2021-11-03 at 4 35 07 PM" src="https://user-images.githubusercontent.com/44172267/140233259-23306558-582a-43c6-afde-7a38aa7acfa3.png">
<img width="647" alt="Screen Shot 2021-11-03 at 4 35 17 PM" src="https://user-images.githubusercontent.com/44172267/140233271-8301bdf6-20d9-4dc8-b5f5-d79e01e2e1db.png">

These color samples need to show different values based on the current theme in `<Sample />`, e.g. Gray 500 shows `#2B2233` in light mode and `#EBE6EF` in dark mode. This means they need to know which theme (`light`/`dark`) has been selected, hence the theme context.

Reply:
Ahh... are we going to have a separate palette for dark and light modes? I think the initial plan was to have a single palette (but not sure if that's still the plan)

Reply:
Yes, the plan is to have separate palettes 🌈